### PR TITLE
Update libmesh

### PR DESCRIPTION
**Pull request by lindsayad**
- Fixes for `--disable-amr`                                                                                               
- Don't leak temporary Netgen `LDFLAGS` into `libmesh_LDFLAGS`                                                            
- Include `<memory>` in `getpot.h`                                                                                        
- Always build all unit tests with plain `make`                                                                           
- Implement improved `ASPECT_RATIO` quality metric for Quad elements                                                      
- Add `Quad::quality(WARP)` implementation                                                                                
- Smoother unit tests                                                                                                     
- Fix `ReplicatedMesh(DistributedMesh &)` construction                                                                    
- Better error when asked to tetrahedralize a broken boundary                                                             
- Add `Elem::edges_adjacent_to_node()` API                                                                                
- Allow all Petsc matrix types to proceed from a common base class                                                        
- Fix bad calls to `LibmeshPetscCall` when we don't have `this->comm()`                                                   
- Call netgen only on rank 0                                                                                              
- Add generic `Elem::quality(MIN,MAX_DIHEDRAL_ANGLE)` implementation                                                      
- Close solution before zeroing for eigen systems                                                                         
- Add matrix iterator APIs to match vector iterator APIs

### CODE REVIEW

**Comment by loganharbour** [comment]
Invalidating everything here (just pushed the openmpi recipe changes now that it's in master) and adding the rest of the next recipes.

**Review by loganharbour — APPROVED** [review]
Providing testing passes, versioning content is good.

File: modules/doc/content/newsletter/2024/2024_08.md [hunk 1]
```diff
@@ -9,6 +9,24 @@ for a complete description of all MOOSE changes.
...
+- Better error when asked to tetrahedralize a broken boundary
+- Add `Elem::edges_adjacent_to_node()` API
+- Allow all Petsc matrix types to proceed from a common base class
+- Fix bad calls to `LibmeshPetscCall` when we don't have `this->comm()`
+- Call netgen only on rank 0
+- Add generic `Elem::quality(MIN,MAX_DIHEDRAL_ANGLE)` implementation
+- Close solution before zeroing for eigen systems
+- Add matrix iterator APIs to match vector iterator APIs
+
+
```
Comment (roystgnr) [inline modules/doc/content/newsletter/2024/2024_08.md:29]:
```suggestion
- Don't leak temporary Netgen `LDFLAGS` into `libmesh_LDFLAGS`.
- Include `<memory>` in `getpot.h` for more compiler compatibility.
- Implemented an improved `ASPECT_RATIO` quality metric for Quad elements.
- Added `Quad::quality(WARP)` implementation.
- Added unit tests for mesh smoothers.
- Fixed `ReplicatedMesh(DistributedMesh &)` construction.
- Report a more useful error message when asked to tetrahedralize a broken boundary.
- Add `Elem::edges_adjacent_to_node()` API
- Allow all Petsc matrix types to proceed from a common base class.
- Call netgen only on rank 0, then broadcast, for better reliability when tetrahedralizing in parallel.
- Add generic `Elem::quality(MIN,MAX_DIHEDRAL_ANGLE)` implementation.
- Close solution before zeroing for eigen systems.
- Add matrix iterator APIs to match vector iterator APIs.
```

**Review by roystgnr — APPROVED** [review]
Suggested tweaks for the newsletter, but those can wait for later.

I'd love to get libMesh/libmesh#3930 in here but if we're almost ready to merge then I'll get it in the next one.

**Comment by roystgnr** [comment]
> Added

Wait, really?  Thanks, but I'd have waited a couple hours to be safe - the devel->master testing in https://civet.inl.gov/event/193352/ isn't done yet.  (though admittedly it's mostly done and I'll be surprised if any of the blocking recipes have a real failure at this point)

**Comment by lindsayad** [comment]
> Thanks, but I'd have waited a couple hours to be safe - the devel->master testing in https://civet.inl.gov/event/193352/ isn't done yet.

I believed in you @roystgnr and I was not disappointed

**Pull request merged**